Router created

EK-machine opened the pull request:
Router was created.
For the moment available only with URL typing in the browser address bar

EK-machine committed:
React import

Imported React from 'react'

EK-machine committed:
Router created

Router was created.
For the moment available only with URL typing in the browser address bar

maksim-vasileuski commented on src/App.js:
```diff
@@ -18,6 +20,9 @@ function App() {
           Learn React
         </a>
       </header>
+      <div>
```
В чем смысл этого дива? На данный момент насколько я понимаю никакой нагрузки не несет.

EK-machine replied:
Создал этот див, чтобы роутер отдельно обернуть в него. Так получается, что роутер бы лежал сразу в диве, в котором сразу целиком лежит всё приложение.
Если это не правильно, то просто сразу помещу элемент <Router /> в див приложения.

maksim-vasileuski replied:
в целом философия такая что у любого элемента должна быть своя цель и функция. если что-то можно убрать и от этого ничего не поменяется - значит надо убрать. чем проще тем лучше
тут на данный момент он ничего не делает. со временем возможно и понадобится какой-то враппер. если будут какие-то стили у него или что-то такое. но заранее точно не надо добавлять

EK-machine replied:
понял теперь о чём ты говорил, исправил

maksim-vasileuski commented on src/router/comment.js:
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
```
папка роутер, не оч название. можешь почитать статеек о том как организовывать код по папкам. 
нужно что-то для переиспользуемых частей типа src/common/components src/common/utils и тд
а то что специфично для страниц надо в папку типа pages. и там pages/home pages/reviews, pages/404

EK-machine replied:
сменил на pages

maksim-vasileuski commented on src/router/error.js:
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Error = ()=>{
```
у тебя скачут пробелы, отступы. сейчас забей, позже наверну тебе линтер и приттиер, чтобы не было такого

EK-machine replied:
ок

maksim-vasileuski commented on src/router/comment.js:
```diff
@@ -0,0 +1,10 @@
+
+const Comment = () =>{
+    return(
+        <div>
```
у тебя везде всё завернуто в дивы. я бы рекомендовал ничего не делать пока оно не понадобится. во всех твоих примерах этот див не меняет ничего. на нем нет никаких стилей, логики тоже. по сути это только усложняет дерево. такого стоит избегать

EK-machine replied:
исправил

maksim-vasileuski commented on src/router/error.js:
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Error = ()=>{
```
Эрор плохое навание для этого компонента, это
Насчет наименования, учитывая что это у тебя контейнеры страниц я бы советовал добавлять тематические окончания. типа - NotFoundPage; 
А также называют файлы с компонентами также как и компонент. Компоненты в реакте принято с большой буквы.
Ну и да, все то что выше тут ко всем файлам относится

EK-machine replied:
исправил

maksim-vasileuski commented on src/router/error.js:
```diff
@@ -0,0 +1,11 @@
+
+const Error = ()=>{
+    return (
+        <div>
```
этот див также избыточен, но он немного особенный т.к реакт не даст тебе экспортировать 2 элемента.
можно вместо дива оборачивать во фрагмент - <></>

EK-machine replied:
исправил

maksim-vasileuski commented on src/router/index.js:
```diff
@@ -0,0 +1,26 @@
+import Comment from './comment';
+import Error from './error';
+
+const ReactRouterSetup = () => {
```
нельзя называть глаголами или действиями. 
если хочется вынести часть с роутером в отдельный компонент, надо смотреть семантически что это такое. Тут это что-то типа main или content. но тут я бы просто все это запихнул в App.js пока что. 

если нет прям уверенности в том как что будет организованно, то разбивать надо по мере того как необходимость и блоки становятся очевидными

EK-machine replied:
переименовал

EK-machine committed:
Promo router corrections

Renamed router folder.
Deleted excess divs at each of the pages.
Renamed the components in accordance with the pages titles.

EK-machine committed:
font awesome

installed font awesome to the project

EK-machine committed:
Styled components

Installed styled components

EK-machine committed:
Merge branch 'master' of https://github.com/EK-machine/promo into master

EK-machine committed:
Merge branch 'master' into Router-creation

EK-machine committed:
Merge branch 'Adding-style-components' into Router-creation

EK-machine committed:
Merge branch 'Adding-font-awesome' into Router-creation

EK-machine committed:
Finished corrections according to eslint rules

EK-machine committed:
Finished corrections according to eslint rules

EK-machine committed:
saved last corrections

maksim-vasileuski commented on public/index.html:
```diff
@@ -10,34 +10,23 @@
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
+    Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will work
```
Перед тем как делать ПР или сдавать его проверку всегда проверяй ручками изменения. тут ты раскомментил какие-то штуки и они пролезли на ЮИ. Это надо удалить. Остальные комменты из этого файла в целом тоже.

maksim-vasileuski commented on src/App.jsx:
```diff
@@ -1,6 +1,13 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+// import styled from 'styled-components';
```
удоли)
если надо будет потом импортировать, потом добавишь. а пока что это мусор

maksim-vasileuski commented on src/pages/comment-page.js:
```diff
@@ -0,0 +1,4 @@
+import React from 'react';
```
У тебя что-то не так с линтером и приттиером. Либо ты повыключал правила нужные. Любой файл содержащий компонент должен быть с расширением jsx. любой файл содержащий компонент пишется с большой буквы и кэмелкейсом

EK-machine replied:
В файле "eslintrc.json" исправил правила: 
1) 
  "react/jsx-filename-extension": [1, { "allow": "as-needed" }]

2) 
 "react/jsx-pascal-case": [1, { "allowAllCaps": false, "ignore": [] }]

Так же переименовал сами файлы и поменял расширения, перепрописал пути (в App.jsx для компонента Router).

maksim-vasileuski commented on src/App.jsx:
```diff
@@ -19,6 +26,7 @@ function App() {
           CI/CD Works!
         </a>
       </header>
+      <Router />
```
Это долно называться не роутер. Роутер это инструмент, а у тебя в этом компоненте весь контент приложения по сути. + роутер может быть использован в других местах тоже.
По сути это центральный блок. можно что-то в духе PageContainer

maksim-vasileuski replied:
Ну и тут я бы схему с индекс файлом не использовал. просто поменял бы имена и явно имортировал то что надо

EK-machine replied:
Исправил. Назвал как router из-за того, что на момент создания это было для меня явным.

EK-machine committed:
Corrected ESlint rules and changed naming of pages

EK-machine committed:
made correction in package json

maksim-vasileuski commented on src/pages/PagesContainer.jsx:
```diff
@@ -0,0 +1,24 @@
+import CommentPage from './CommentPage';
+import PageNotFound from './PageNotFound';
+
+const PromoRouter = () => (
```
у тебя файл называется PagesContainer а компонент внутри PromoRouter. надо бы это дело синхронизировать

EK-machine replied:
готово

maksim-vasileuski commented on src/pages/PageNotFound.jsx:
```diff
@@ -0,0 +1,9 @@
+    <h4>Here is an error</h4>
+  </>
+);
+export default PageNotFound;
```
все страницы называются чтоТоТамPage а эта PageЧтоТоТам. переименуй наверное. Page в конце это как суффикс уточняющий.

EK-machine replied:
Готово

maksim-vasileuski reviewed (changes requested):
Almost there. Пара минорных моментов, и можно мержить

EK-machine committed:
made correction in package json
